QgisModelBaker/libqgsprojectgen/generator/generator.py

```diff
@@ -101,12 +101,19 @@ def layers(self, filter_layer_list=[]):
             if filter_layer_list and record['tablename'] not in filter_layer_list:
                 continue
 
-            alias = record['table_alias'] if 'table_alias' in record else ''
             is_domain = record['kind_settings'] == 'ENUM' or record[
                 'kind_settings'] == 'CATALOGUE' if 'kind_settings' in record else False
             is_structure = record['kind_settings'] == 'STRUCTURE' if 'kind_settings' in record else False
             is_nmrel = record['kind_settings'] == 'ASSOCIATION' if 'kind_settings' in record else False
 
+            alias = record['table_alias'] if 'table_alias' in record else None
+            if not alias:
+                if is_domain:
```

Comment:
I'm not sure of this approach. Take for example this ili_name from one of our domains (obtained from `t_ili2db_classname`):

`Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0.LC_PuntoTipo`

Where: 

`Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0`: Model name
`LC_PuntoTipo`: Domain table name

Taking the last part of the split array could be the way to go here too.

Perhaps you're referring to this ili_name (obtained from `t_ili2db_attrname`)?
`Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0.Levantamiento_Catastral.LC_PuntoControl.PuntoTipo`

Which is the iliname of the attribute that has been defined to store domain values.

Reply:
Hm... okay. I assume the sql name of the created table is lc_puntotipo then?

On road simple is that the domain table containing the enumerations is called landcover_type,  because it containes LandCover.Type. So I thought this is the way it works. 

Can you provide me the model so I can see if I find a better solution?

Reply:
> Hm... okay. I assume the sql name of the created table is lc_puntotipo then?

Right.

> Can you provide me the model so I can see if I find a better solution?

Sure, you can find it here: https://github.com/SwissTierrasColombia/LADM_COL/releases/download/1.0/Modelo_Aplicacion_LADMCOL_Levantamiento_Catastral_V1_0.zip

The model is called `Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0` and an example of a domain is: `Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0.LC_PuntoTipo`.

Reply:
Where can I find the Swiss model you're using? It would be good to see if there are differences in the modelling itself.

Reply:
https://www.interlis.ch/download/interlis2/Testbeispiel_20160812.zip <- this here.

And we are using it in the tests for Model Baker as well.

Reply:
There is sometimes a `Topic` that it integrated into the ili-name but sometimes there isn't.

Reply:
Is it possible that there are duplicate names and it adds the Topic when there is a name collision without the topic?

Reply:
In the case of the simple road model there would be duplicates if there are no topics. But I don't know if it does not take the topic if there are no duplicates. I would need to check.

Reply:
Sorry what I said here https://github.com/opengisch/QgisModelBaker/pull/423#discussion_r478205903 is wrong: There are no duplicates without the topics. The topics are not integrated into to sql-name.

When the iliname is:
`<MODEL>.<TOPIC>.<CLASS>` it creates the sqlname `<CLASS>`
When the iliname is:
`<MODEL>.<CLASS>` it creates the sqlname `<CLASS>`

But I think I compared not the same things above:

While `Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0.LC_PuntoTipo` is under `DOMAIN` in the ili file of Germán, the  `RoadsSimple.Roads.LandCover.Type` is defined as this:
```
    CLASS LandCover =
      Type: MANDATORY (
        building,
        street,
        water,
        other);
      Geometry: MANDATORY SURFACE WITH (STRAIGHTS)
        VERTEX Point2D WITHOUT OVERLAPS > 0.100;
    END LandCover;
```
So we have to find out where to get the difference from then...

Reply:
What I was referring to is https://github.com/claeis/ili2db/blob/master/docs/ili2db.rst

> Option | Beschreibung
> -- | --
> `--disableNameOptimization` | Schaltet die Nutzung von unqualifizierten Klassennamen aus. Für alle  Tabellennamen werden qualifizierte Interlis-Klassennamen  (Model.Topic.Class) verwendet (und in einen gültigen Tabellennamen  abgebildet). (siehe Kapitel Abbildungsregeln/Namenskonventionen)
> `--nameByTopic` | Für alle Tabellennamen werden teilweise qualifizierte  Interlis-Klassennamen (Topic.Class) verwendet (und in einen gültigen  Tabellennamen abgebildet). (siehe Kapitel  Abbildungsregeln/Namenskonventionen)

We have some control over the sql name generation. But it's probably not related to this discussion.

Reply:
> While Modelo_Aplicacion_LADMCOL_Lev_Cat_V1_0.LC_PuntoTipo is under DOMAIN in the ili file of Germán, the RoadsSimple.Roads.LandCover.Type is defined as this:

It's an in-line ENUM. We're moving forward.

Proposal:

1. If it's a class and does not have alias: Take the last part from its iliname. 
2. If it's a domain and does not have alias:
   2.a. Check if its `iliname` is also an iliname in `t_ili2db_attrname`, which indicates we have an in-line ENUM. If so, to obtain a proper alias, take the last 2 parts of the iliname, and join them with an underscore (e.g., `LandCover_Type`).
   2.b. If `iliname` is not in `t_ili2db_attrname`, to get a proper alias take the last part of `iliname`, as in the table case (e.g., `LC_PuntoTipo`).

Note: 2.a. is ugly, I know. Perhaps we could consider whether it makes sense to ask Claude to make a difference between in-line ENUMs and non-in-line ENUMs in the `t_ili2db_table_prop` table.

Reply:
Just for information regarding https://github.com/opengisch/QgisModelBaker/pull/423#discussion_r478579452 of @m-kuhn : In this case we would have roadssimpleroads_landcover on `--disableNameOptimization` and roads_landcover on `--nameByTopic`

@gacarrillor I don't think it's that ugly. It checks if it's an ENUM and if this ENUM is an attribute (in-line) then it needs to contain the name of the owner table in the table name. I think it's a straight forward solution.